eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/internal/SetIterables.java

```diff
@@ -51,8 +52,16 @@ private SetIterables()
             SetIterable<? extends E> setB,
             R targetSet)
     {
-        Iterate.addAllIterable(setA, targetSet);
-        Iterate.addAllIterable(setB, targetSet);
+        if (setA.size() > setB.size())
```

Comment:
the order doesn't seem to make any difference here...

Reply:
Yes, I changed back to optimizing `union` so I can use `UnifiedSet.newSet()`, and reverted `unionInto`.